### PR TITLE
Extract `Brewfile` locator logic to its own module

### PR DESCRIPTION
**Pull request by aramprice**
`Bundle.brewfile` and `Dumper.dump_brewfile` contained duplicate logic
for locating the `Brewfile`. This logic now resides in the `Brewfile`
module.

This is the first step in addressing https://github.com/Homebrew/homebrew-bundle/issues/368

### CODE REVIEW

**Comment by aramprice** [comment]
Travis appears to be failing on master so I'm not sure what to expect from the automated checks.

Locally I have the following:

```
→ bundle exec rake && brew style homebrew/bundle
/System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/bin/ruby -I/Users/aram/workspace/homebrew-bundle/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.8.0/lib:/Users/aram/workspace/homebrew-bundle/vendor/bundle/ruby/2.3.0/gems/rspec-support-3.8.0/lib /Users/aram/workspace/homebrew-bundle/vendor/bundle/ruby/2.3.0/gems/rspec-core-3.8.0/exe/rspec --pattern spec/\*\*\{,/\*/\*\*\}/\*_spec.rb
............................................................................................................................................................................

Finished in 6.06 seconds (files took 0.30536 seconds to load)
172 examples, 0 failures

Coverage report generated for RSpec to /Users/aram/workspace/homebrew-bundle/coverage. 1898 / 1898 LOC (100.0%) covered.

57 files inspected, no offenses detected
```

File: lib/bundle/brewfile.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "English"
```
Comment (colindean) [inline lib/bundle/brewfile.rb]: TIL https://ruby-doc.org/stdlib-2.3.7/libdoc/English/rdoc/English.html

**Comment by colindean** [comment]
I'd like to wait until Travis catches up before merging this but otherwise 👍 

Thanks for the refactor!

**Comment by MikeMcQuaid** [comment]
> Travis appears to be failing on master so I'm not sure what to expect from the automated checks.

Sorry for that. Let's wait until it's resolved and CI rerun before this is merged.

**Comment by MikeMcQuaid** [comment]
@aramprice Please rebase on `master` and CI should be 💚.

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @aramprice!

**Pull request merged**